source/extensions/filters/http/decompressor/decompressor_filter.h

```diff
@@ -100,9 +100,18 @@ class DecompressorFilterConfig {
   const std::string& contentEncoding() { return decompressor_factory_->contentEncoding(); }
   const RequestDirectionConfig& requestDirectionConfig() { return request_direction_config_; }
   const ResponseDirectionConfig& responseDirectionConfig() { return response_direction_config_; }
+  // TODO: replace std::vector for std::pair. For some reason the macro is failing to build with a
+  // multivariate templated type.
+  const std::vector<Http::LowerCaseString>& trailersStrings() const {
+    CONSTRUCT_ON_FIRST_USE(
+        std::vector<Http::LowerCaseString>,
+        {Http::LowerCaseString(fmt::format("{}-compressed-bytes", trailers_prefix_)),
+         Http::LowerCaseString(fmt::format("{}-uncompressed-bytes", trailers_prefix_))});
+  }
```

Comment:
It's still unclear to me why one function returning a vector (or a pair) is preferable over two functions returning a single string, e.g. `const Http::LowerCaseString& trailerCompressedBytesString() const` and `const Http::LowerCaseString& 
 trailerUncompressedBytesString() const`.

Is it an optimization? Or
```
CONSTRUCT_ON_FIRST_USE(Http::LowerCaseString, fmt::format("{}-uncompressed-bytes", trailers_prefix_))
```
fails to build?

Reply:
I want it to keep it to one function. But I agree, ultimately, that this is better than a vector given the pair does not work. Updated.